src/htsp_server.c

```diff
@@ -1019,82 +1010,66 @@ htsp_build_dvrentry(htsp_connection_t *htsp, dvr_entry_t *de, const char *method
-              }//END got an imagecache location
-            }//END icon not null
-
-            //The authority and country are added for Kodi's parentalRatingSource field.
-            //Kodi looks for the authority first and if that is not present, then it uses the country.
-            //There could be no label, but there could be an age.  The authority &
-            //country could still be useful.
-            if (htsp->htsp_version > 40){
-                if(de->de_rating_label->rl_authority){
-                  htsmsg_add_str(out, "ratingAuthority", de->de_rating_label->rl_authority);
-                }//END authority saved not null
-
-                if(de->de_rating_label->rl_country){
-                  htsmsg_add_str(out, "ratingCountry", de->de_rating_label->rl_country);
-                }//END country saved not null
-            }//END ver > 40
-          }//END rating label not null
-        }//END this is a scheduled recording.
-        else
-        {
-          if(de->de_rating_label_saved){
-            if(de->de_rating_label_saved){
-              htsmsg_add_str(out, "ratingLabel", de->de_rating_label_saved);
-            }
-            if(de->de_rating_icon_saved){
-              str = de->de_rating_icon_saved;
-              if (!strempty(str)) {
-                str = imagecache_get_propstr(str, buf, sizeof(buf));
-                if (str)
-                  htsmsg_add_str(out, "ratingIcon", str);
-              }//END got an imagecache location
-            }//END icon not null
-
-            if (htsp->htsp_version > 40){
-              if(de->de_rating_authority_saved){
-                htsmsg_add_str(out, "ratingAuthority", de->de_rating_authority_saved);
-              }//END authority saved not null
-
-              if(de->de_rating_country_saved){
-                htsmsg_add_str(out, "ratingCountry", de->de_rating_country_saved);
-              }//END country saved not null
-            }//END version > 40
+    //Only add the rating label fields if rating labels are enabled.
+    if(epggrab_conf.epgdb_processparentallabels){
+      //If this is still scheduled (in the future) then send the current values,
+      //if not, send the 'saved' values.
 
+      if(de->de_sched_state == DVR_SCHEDULED){
+        if(de->de_rating_label){
+          if(de->de_rating_label->rl_display_label){
+            htsmsg_add_str(out, "ratingLabel", de->de_rating_label->rl_display_label);
           }
-        }//END this is not a scheduled recording.
-      }//END processing rating labels is enabled
-    }
+          //If the rating icon is not null.
+          if(de->de_rating_label->rl_icon){
+            str = de->de_rating_label->rl_icon;
+            if (!strempty(str)) {
+              str = imagecache_get_propstr(str, buf, sizeof(buf));
+              if (str)
+                htsmsg_add_str(out, "ratingIcon", str);
+            }//END got an imagecache location
+          }//END icon not null
```

Comment:
As a general side note: If the respective code blocks are short, imo these comments do not add any value as you always have the related `if` /`else`on the screen. But you always risk that the comments get outdated whenever you change the content of the code block. I suggest to just remove these comments.

OTOH; if the code blocks are so long that you do not see the conditions on your screen when looking at the end of the block, then this is a clear indicator that the block is to long. You should factor out (parts of) the code in the block to separate helper functions with speaking names. So you get readability of the code without the price for the comments explaining which block end we're looking at.